packages/core/src/routes/applications/application.ts

```diff
@@ -306,6 +306,10 @@ export default function applicationRoutes<T extends AuthedRouter>(
     }),
     async (ctx, next) => {
       const { id } = ctx.guard.params;
+      const { type, protectedAppMetadata } = await findApplicationById(id);
+      if (type === ApplicationType.Protected && protectedAppMetadata) {
+        await protectedApps.deleteRemoteAppConfigs(protectedAppMetadata.host);
```

Comment:
double-check `protectedAppMetadata.host` is used as the unique identifier.

Reply:
Yes